### PR TITLE
Add missing code to reset RequestEventDispatchers when used in a pool

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
I was originally viewing this as a bug fix for v4.1.x, so didn't include release notes. Then I thought it's more like a feature PR. What do people think?

**Comment by nemanja-boric-sociomantic** [comment]
I think this should go as a bugfix. Otherwise, no real implementation is able to use the facility, which is included in v4.1.x, and that was not our intention.

**Comment by gavin-norman-sociomantic** [comment]
Yeah that was my initial thought, until I noticed that it's actually just pure new features. I'll rebase it.

File: src/swarm/neo/request/RequestEventDispatcher.d [hunk 1]
```diff
@@ -235,6 +235,24 @@ public struct RequestEventDispatcher
 
     /***************************************************************************
 
+        Resets this instance to its initial state. This method should be called
+        when retriving an instance from a pool or free list.
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/request/RequestEventDispatcher.d]: typo `retriving`

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/request/RequestEventDispatcher.d]: Updated.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**